main.tf

```diff
@@ -33,6 +33,10 @@ module "vpc" {
   single_nat_gateway     = true                                        // Use a single NAT gateway as that's the simplest and also all we need.
   tags                   = var.tags                                    // Use the tags specified as a variable.
 
+  enable_ipv6                                    = var.enable_ipv6 // this will provide Amazon-provided IPv6 CIDR block which is a /56 block
+  public_subnet_assign_ipv6_address_on_creation  = var.enable_ipv6 // this will help the EC2 to get the IPV6 address when it boots
+  private_subnet_assign_ipv6_address_on_creation = var.enable_ipv6 // this will help the EC2 to get the IPV6 address when it boots
+
   // Create one private subnet per AZ (e.g. "10.1.0.0/24", "10.1.1.0/24", "10.1.2.0/24", ...).
```

Comment:
Don't we need the same logic for IPv6? Either for private and public in case unique local IPv6 is used or only for public when global unicast IPv6 IPs are used. If not, how are IPv6 CIDRs assigned to AZs?

Reply:
we can have the option to ask the users to configure the ipv6 ip block but it's not necessary since there are too many ipv6 address, you can simply ask the AWS to give you a /56 public ipv6 for your VPC.

you will need to bring your ipv6 or configure extra resources if you want to go with the other options as the following pic shows.

My comment also indicates that it will just use AWS-provided CIDR ipv6 block.
```  enable_ipv6 = var.enable_ipv6 // this will provide Amazon-provided IPv6 CIDR block which is a /56 block```

![image](https://github.com/isovalent/terraform-aws-vpc/assets/8106123/a04cb1a1-588e-4c31-9b8d-943131bbac65)

https://github.com/isovalent/terraform-aws-vpc/pull/17/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR70-R79

this part will ensure how subnets get the IPV6 address from the block. it looks weird at the first glance. in fact it will add 8 bits by default and with your offset to assign the CIDR to the subnets

for example if you have CIDR block 2002::/56. this will assign the cidr to the private subnets 2002:0000:0000:0000::/64
2002:0000:0000:0001::/64
2002:0000:0000:0002::/64

and public subnets like this 

2002:0000:0000:000a::/64
2002:0000:0000:000b::/64
2002:0000:0000:000c::/64
2002:0000:0000:000d::/64


Here is the reference that I use 

https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/examples/ipv6-dualstack/main.tf